lib/puppet/node.rb

```diff
@@ -27,7 +27,10 @@ def initialize_from_hash(data)
     @name       = data['name']       || (raise ArgumentError, _("No name provided in serialized data"))
     @classes    = data['classes']    || []
     @parameters = data['parameters'] || {}
-    @environment_name = data['environment']
+    env_name = data['environment']
+    env_name = env_name.intern unless env_name.nil?
+    @environment_name = env_name
+    environment = env_name
```

Comment:
I haven't been able to find any situation when this fails. All tests pass, both the rspec tests and the acceptance tests (aside from the acceptance tests that failed due to problems with `ci:test:git`).